src/ui/storyTellingServerUI.js

```diff
@@ -0,0 +1,175 @@
+ */
+sjrk.storyTelling.loadStoryFromParameter = function (theme, options) {
+    var storyPromise = fluid.promise();
+
+    var storyId = sjrk.storyTelling.getParameterByName("id");
+
+    if (storyId) {
+        $.get("/stories/" + storyId, function (data) {
+            var retrievedStory = JSON.parse(data);
+
+            options = options || {};
+            options.distributeOptions = {
+                "target": "{that story}.options.model",
+                "record": retrievedStory
+            };
+
+            var storyViewComponent;
+            if (theme === "base") {
+                storyViewComponent = sjrk.storyTelling.page.storyView(options);
+            } else {
+                storyViewComponent = sjrk.storyTelling[theme].storyView(options);
+            }
+
+            storyPromise.resolve(storyViewComponent);
+        }).fail(function (jqXHR, textStatus, errorThrown) {
+            storyPromise.reject({
+                isError: true,
+                message: errorThrown
+            });
+        });;
+    } else {
+        storyPromise.reject({
+            isError: true,
+            message: "No story ID provided"
+        });
+    }
+
+    return storyPromise;
+};
+
+/* Loads a story Browse page and populates it with a set of stories
+ * - "theme": the theme of the story Browse page (pass "page" in for base theme)
+ * - "options": additional options to merge into the Browse page
+ */
+sjrk.storyTelling.loadBrowse = function (theme, options) {
+    var storiesPromise = fluid.promise();
+
+    $.get("/stories", function (data) {
+        var browseResponse = JSON.parse(data);
+
+        options = options || {};
+        options.distributeOptions = {
+            "target": "{that storyBrowser}.options.model",
+            "record": browseResponse
+        };
+
+        var storyBrowseComponent;
+        if (theme === "base") {
+            storyBrowseComponent = sjrk.storyTelling.page.storyBrowse(options);
+        } else {
+            storyBrowseComponent = sjrk.storyTelling[theme].storyBrowse(options);
+        }
+
+        storiesPromise.resolve(storyBrowseComponent);
+    }).fail(function (jqXHR, textStatus, errorThrown) {
+        storiesPromise.reject({
+            isError: true,
+            message: errorThrown
+        });
+    });
+
+    return storiesPromise;
+};
+
+/* Loads custom theme files if and only if one of either the theme override or
+ * server configuration theme are present AND not set to the base theme.
+ * Once complete, the provided callback function is called and the theme name is
+ * passed into it. If custom theme files are not loaded, the "base" theme is passed.
+ * - "callback": a function to call once everything has completed. Will be called
+ *               regardless of whether theme information was specified or retrieved
+ * - "themeOverride": allows overriding of the theme stored in the configuration
+ */
+sjrk.storyTelling.loadThemedPage = function (callback, themeOverride) {
```

Comment:
It's weird to accept "themeOverride" parameter that could override the theme sent back from the server, which is the theme defined in the config file. As what's shown in your test case, what would happen if this value is provided as `null` or `undefined`? Does the client side have to perform its own fall back to the base? 

Could you elaborate the purpose of "themeOverride" parameter? Thanks.

Reply:
I had added that in order to do more robust testing of the `loadThemedPage` function, since it's a browser test and not based on varying server config files. This is the only usage of it that parameter that I've anticipated. If `themeOverride` is falsy, it will fall back to the theme returned from the request to `/clientConfig`, according to line 118.

I suppose the structure of this function should be reworked, if `themeOverride` is going to be kept, since some of the possible cases are not handled, namely if the request fails but `themeOverride` is truthy, or if the request fails and `themeOverride` is truthy but also not valid.

Reply:
A better way to test it is to create a mocking server response for /clientConfig, which mocks http responses you are expecting for your test cases. Infusion uses "Sinon.js" library that also has [the feature of creating a fake server](https://sinonjs.org/#fake-server).